content/components/dialog.mdx

```diff
@@ -0,0 +1,383 @@
+  <div>
+    <img
+      width="650"
+      alt=""
+      src="https://user-images.githubusercontent.com/18661030/185506131-1436b06b-a8fb-4de8-b9b2-4715ab0a63d2.png"
+    />
+    <Caption>Body content will scroll vertically if the content exceeds the overall Dialog height.</Caption>
+  </div>
+  <div>
+    <img
+      width="650"
+      alt=""
+      src="https://user-images.githubusercontent.com/18661030/186003712-36441f05-1407-4b03-a50a-3773275e4d61.png"
+    />
+    <Caption>The body may contain additional content such as banners and form controls.</Caption>
+  </div>
+</Box>
+
+### Footer region
+
+The footer region may be used to show confirmation actions, navigation links, or specialized actions. Primary actions should be aligned to the right of the footer, and grouped why additional related actions. Secondary actions may be shown aligned to the left side.
+
+If the content area has overflow scrolling, a divider will be shown between the footer and content area. Otherwise, showing a divider is optional.
+
+<img
+  src="https://user-images.githubusercontent.com/293280/175850630-9f3df284-c69b-4d16-9e57-ea04a0202d04.png"
+  width="960"
+  alt=""
+/>
+<Caption>Use buttons in a footer to let the user complete a task.</Caption>
+
+<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={5}>
+  <div>
+    <img
+      src="https://user-images.githubusercontent.com/293280/175850639-a77d537e-2807-47d2-b456-a298d5f44918.png"
+      alt=""
+    />
+    <Caption>Footer with small buttons aligned to the right.</Caption>
+  </div>
+  <div>
+    <img
+      src="https://user-images.githubusercontent.com/293280/175850637-3b193552-b35f-4b07-b2d6-4ca546c8131a.png"
+      alt=""
+    />
+    <Caption>Footer with a secondary button aligned to the left.</Caption>
+  </div>
+</Box>
+
+### Backdrop
+
+Dialog displays with a **backdrop**, which dims the rest of the page. The backdrop is visible when the Dialog is centered on the page, or attached to a side as a sheet.
+
+By default, clicking on the **backdrop** will dismiss the Dialog. However, if the Dialog contains a form that can have unsaved changes, the **backdrop** won't dismiss the Dialog, regardless of the state of the form.
+
+### Dimensions
+
+Dialogs should never extend beyond the viewport sizes, and may resize to fit smaller devices. The body contents should fit narrow viewports with a minimum width of `320px`. If the contents extend vertically beyond the specified height, the **body** region will scroll.
+
+The `height` and `width` may be set to `auto` which will set a minimum width of `192px` and respond to the contents of the Dialog. Additionally, an explicit `height` and `width` may be set with a specific set of options.
+
+<img
+  width="1292"
+  alt=""
+  src="https://user-images.githubusercontent.com/18661030/185178528-e1c176ef-f407-40ab-8729-965e21d64f6d.png"
+/>
+<Caption>Recommended height and width combinations for Dialog</Caption>
```

Comment:
+1 to the table or even a legend with all the possible size combinations:

Other thoughts on this regarding this [comment](https://github.com/primer/design/pull/281#discussion_r949690844):
- 👍 adding a width/height label to the X/Y axis
- 👍 to use a single purple color rather than the sequential palette. It's clearer when looking for what will be a good or a bad combination.
- Should we explain why we provide these combinations and not others? e.g, line length good practices